src/utils.cpp

```diff
@@ -10,7 +10,7 @@ vector<unsigned int> intsPerFloor;
  * input_moduli, from the given file.
  */
 void read_moduli_from_csv( \
-        string filename, vector<mpz_class> *moduli, vector<int>*IDs) {
+        string filename, string base, vector<mpz_class> *moduli, vector<int>*IDs) {
```

Comment:
I propose the following signature:
```
void read_moduli_from_csv(string filename, vector<mpz_class> *moduli, int base=16)
```
So that the logic from `main` is simpler.

Reply:
Done